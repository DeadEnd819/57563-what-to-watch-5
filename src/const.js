export const BACKEND_URL = `https://5.react.pages.academy/wtw`;
export const REQUEST_TIMEOUT = 5000;

export const RATINGS_MOVIE_STARS = [`1`, `2`, `3`, `4`, `5`];
export const MORE_MOVIE_COUNT = 4;
export const SHOW_MOVIE_COUNT = 8;
export const MIN_STARRING_COUNT = 0;
export const MAX_STARRING_COUNT = 4;
export const VIDEO_TIME_OUT = 1000;
export const MAX_GENRES_COUNT = 9;
export const DEFAULT_MOVIE_RATING = `3`;
export const DEFAULT_REVIEW_TEXT = ``;

export const HttpCode = {
  SUCCESS: 200,
  UNAUTHORIZED: 401
};

export const AuthorizationStatus = {
  AUTHORIZED: `AUTHORIZED`,
  NOT_AUTHORIZED: `NOT_AUTHORIZED`,
};

export const ActionType = {
  CHANGE_GENRE: `CHANGE_GENRE`,
  INCREMENT_SHOW_FILMS_COUNT: `INCREMENT_SHOW_FILMS_COUNT`,
  RESET_FILMS_FILTER: `RESET_FILMS_FILTER`,
  LOAD_FILMS: `LOAD_FILMS`,
  LOAD_PROMO_FILM: `LOAD_PROMO_FILM`,
  LOAD_CURRENT_FILM: `CHANGE_CURRENT_FILM`,
  LOAD_REVIEWS: `LOAD_REVIEWS`,
  REQUIRE_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SAVE_AUTHORIZATION_INFO: `SAVE_AUTHORIZATION_INFO`,
  REDIRECT_TO_ROUTE: `REDIRECT_TO_ROUTE`,
  SET_REVIEW_STATUS: `SET_REVIEW_STATUS`,
};

export const ReviewStatus = {
  REVIEW_UPDATED: `REVIEW_UPDATED`,
  REVIEW_NOT_UPDATED: `REVIEW_NOT_UPDATED`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  REVIEW: `/review`,
  PLAYER: `/player`,
};

export const APIRoute = {
  FILMS: `/films`,
  PROMO: `/films/promo`,
  FAVORITE: `/favorite`,
  COMMENTS: `/comments`,
  LOGIN: `/login`,
};

export const TabNames = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `Reviews`
};

export const RatingOptions = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};

export const GenresNames = {
  ALL_GENRES: `All genres`,
};

export const MovieDetailsName = {
  DIRECTOR: `Director`,
  STARRING: `Starring`,
  RUN_TIME: `Run Time`,
  GENRE: `Genre`,
  RELEASED: `Released`,
};

export const ReviewLengths = {
  MIN: 50,
  MAX: 400,
};

export const defaultFilm = {
  id: 1,
  name: ``,
  posterImage: ``,
  previewImage: ``,
  backgroundImage: ``,
  backgroundColor: ``,
  previewVideoLink: ``,
  description: ``,
  rating: 0,
  scoresCount: 0,
  director: ``,
  starring: [``],
  runTime: 0,
  genre: ``,
  released: 0,
  isFavorite: false,
};

export const defaultReview = {
  id: 1,
  user: {
    id: 1,
    name: ``,
  },
  rating: 1,
  comment: ``,
  date: ``,
};

