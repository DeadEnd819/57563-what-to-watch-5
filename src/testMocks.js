export const promoFilm = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  genre: `Comedy`,
  released: 2014,
  isFavorite: false,
};

export const film = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  previewImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  runTime: 99,
  genre: `Comedy`,
  released: 2014,
  isFavorite: true,
};

export const reviewOne = {
  id: 1,
  user: {
    id: 4,
    name: `Kate Muir`,
  },
  rating: 8.9,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`,
};

export const reviewTwo = {
  id: 2,
  user: {
    id: 5,
    name: `Paula Fleri-Soler`,
  },
  rating: 9.1,
  comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  date: `2019-08-08T17:40:56.569Z`,
};

export const reviews = [reviewOne, reviewTwo];

export const genresList = [
  `Action`,
  `Adventure`,
  `Comedy`,
  `Crime`,
  `Drama`,
  `Fantasy`,
  `Thriller`
];

export const state = {
  DATA: {
    films: [film, film, film, film, film, film, film],
    promo: promoFilm,
    currentFilm: film,
    currentReviews: reviews,
  },
  FILTER: {
    genre: `Drama`,
    showFilmsCount: 8,
  },
  USER: {
    name: `name`,
    avatar: `avatar.jpg`,
    status: `NOT_AUTHORIZED`,
    reviewStatus: `REVIEW_NOT_UPDATED`,
  },
};
