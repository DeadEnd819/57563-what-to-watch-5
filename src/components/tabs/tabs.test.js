import React from "react";
import renderer from "react-test-renderer";
import Tabs from "./tabs";
import {film, reviews} from "../../testMocks";
import {TabNames} from "../../const";

jest.mock(`../movie-overview/movie-overview.jsx`, () => `MovieOverview`);
jest.mock(`../movie-details/movie-details.jsx`, () => `MovieDetails`);
jest.mock(`../movie-reviews/movie-reviews.jsx`, () => `MovieReviews`);
jest.mock(`../tabs-item/tabs-item.jsx`, () => `TabsItem`);

const noop = ()=>{};

describe(`Render Tabs`, () => {
  it(`Should Tabs details render correctly`, () => {
    const tree = renderer
      .create(
          <Tabs
            film={film}
            reviews={reviews}
            activeTab={TabNames.DETAILS}
            onTabClick={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Tabs reviews render correctly`, () => {
    const tree = renderer
      .create(
          <Tabs
            film={film}
            reviews={reviews}
            activeTab={TabNames.REVIEWS}
            onTabClick={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Tabs default render correctly`, () => {
    const tree = renderer
      .create(
          <Tabs
            film={film}
            reviews={reviews}
            activeTab={`Tab default`}
            onTabClick={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
