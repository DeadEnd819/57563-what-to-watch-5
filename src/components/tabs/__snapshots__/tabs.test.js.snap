// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Tabs Should Tabs default render correctly 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <TabsItem
        activeTab="overview"
        onTabClick={[Function]}
        tab="overview"
      />
      <TabsItem
        activeTab="overview"
        onTabClick={[Function]}
        tab="details"
      />
      <TabsItem
        activeTab="overview"
        onTabClick={[Function]}
        tab="Reviews"
      />
    </ul>
  </nav>,
  <MovieOverview
    film={
      Object {
        "backgroundColor": "#ffffff",
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
        "director": "Wes Andreson",
        "genre": "Comedy",
        "id": 1,
        "isFavorite": true,
        "name": "The Grand Budapest Hotel",
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "bg-the-grand-budapest-hotel.jpg",
        "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
        "rating": 8.9,
        "released": 2014,
        "runTime": 99,
        "scoresCount": 240,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
          "Saoirse Ronan",
        ],
      }
    }
  />,
]
`;

exports[`Render Tabs Should Tabs details render correctly 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <TabsItem
        activeTab="overview"
        onTabClick={[Function]}
        tab="overview"
      />
      <TabsItem
        activeTab="overview"
        onTabClick={[Function]}
        tab="details"
      />
      <TabsItem
        activeTab="overview"
        onTabClick={[Function]}
        tab="Reviews"
      />
    </ul>
  </nav>,
  <MovieOverview
    film={
      Object {
        "backgroundColor": "#ffffff",
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
        "director": "Wes Andreson",
        "genre": "Comedy",
        "id": 1,
        "isFavorite": true,
        "name": "The Grand Budapest Hotel",
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "bg-the-grand-budapest-hotel.jpg",
        "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
        "rating": 8.9,
        "released": 2014,
        "runTime": 99,
        "scoresCount": 240,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
          "Saoirse Ronan",
        ],
      }
    }
  />,
]
`;

exports[`Render Tabs Should Tabs reviews render correctly 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <TabsItem
        activeTab="overview"
        onTabClick={[Function]}
        tab="overview"
      />
      <TabsItem
        activeTab="overview"
        onTabClick={[Function]}
        tab="details"
      />
      <TabsItem
        activeTab="overview"
        onTabClick={[Function]}
        tab="Reviews"
      />
    </ul>
  </nav>,
  <MovieOverview
    film={
      Object {
        "backgroundColor": "#ffffff",
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
        "director": "Wes Andreson",
        "genre": "Comedy",
        "id": 1,
        "isFavorite": true,
        "name": "The Grand Budapest Hotel",
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "bg-the-grand-budapest-hotel.jpg",
        "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
        "rating": 8.9,
        "released": 2014,
        "runTime": 99,
        "scoresCount": 240,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
          "Saoirse Ronan",
        ],
      }
    }
  />,
]
`;
